app/services/calculators/multiples/same_proceedings.rb

```diff
@@ -18,9 +18,42 @@ def spent_date
       private
 
       def expiry_dates
-        @_expiry_dates ||= disclosure_checks.map(
-          &method(:expiry_date_for)
-        )
+        expiration_dates
+      end
+
+      def expiration_dates
+        return expiry_date_for_all_sentences if all_are_relevant_orders? || no_relevant_orders?
```

Comment:
Could all this be simplified by basically obtaining a collection without the relevant orders, or, if empty, defaulting to all sentences, and then using the resulting dates to pick the max?

Kind of:

```ruby
def expiration_dates
  (sentences_without_relevant_orders.presence || disclosure_checks.presence).map(&method(:expiry_date_for))
end
```